### PR TITLE
[SYCL][ESIMD] Replace std::max_align_t with 16 for overaligned

### PR DESCRIPTION
**Pull request by vasilytric**
Because of the fact that long double is
not a native type in Intel GPUs. So 16 is not driven by any type, but
rather the "oword alignment" requirement for all block loads. In that
sense, std::max_align_t gives wrong idea.

### CODE REVIEW

File: SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp [hunk 1]
```diff
@@ -110,12 +110,13 @@ struct vector {
...
-  template <typename, int> static size_t get_size() {
-    return alignof(std::max_align_t);
-  }
-  static constexpr auto get_value() {
-    return overaligned<alignof(std::max_align_t)>;
-  }
+  // Use 16 instead of std::max_align_t because of the fact that long double is
+  // not a native type in Intel GPUs. So 16 is not driven by any type, but
+  // rather the "oword alignment" requirement for all block loads. In that
+  // sense, std::max_align_t gives wrong idea.
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp]:
```suggestion
  // sense, std::max_align_t would give wrong idea.
```

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp]: this comment was applied in this commit: 5a86f5ce550aeb420a3550b2edca451caa75d1cc

**Comment by vasilytric** [comment]
@kbobrovs Would you mind to take a look?

File: SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp [hunk 2]
```diff
@@ -110,12 +110,13 @@ struct vector {
...
-    return alignof(std::max_align_t);
-  }
-  static constexpr auto get_value() {
-    return overaligned<alignof(std::max_align_t)>;
-  }
+  // Use 16 instead of std::max_align_t because of the fact that long double is
+  // not a native type in Intel GPUs. So 16 is not driven by any type, but
+  // rather the "oword alignment" requirement for all block loads. In that
+  // sense, std::max_align_t would give wrong idea.
+  template <typename, int> static size_t get_size() { return alignof(16); }
```
Comment (kbobrovs) [inline SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp]: Please replace 16 with some descriptive constant. e.g. `oword_align` or similar

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp]: I've done it in this commit: 1782977a21e51d37f123970a3a232bbf88087c65

**Comment by kbobrovs** [comment]
I don't see any changes in the "Files changed" tab.

**Comment by vasilytric** [comment]
> I don't see any changes in the "Files changed" tab.

@kbobrovs This changed has been re-written after merging intel into this branch
Now all changed can be found in this commit: 0704c9b707a8f7af205327f64f71686957eb6a58

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**